src/eip7594/eip7594.c

```diff
@@ -454,6 +454,125 @@ static C_KZG_RET compute_r_powers_for_verify_cell_kzg_proof_batch(
+    fr_t *aggregated_interpolation_poly = NULL;
+    bool *is_cell_used = NULL;
+
+    // Allocate memory
+    ret = new_fr_array(&aggregated_column_cells, FIELD_ELEMENTS_PER_EXT_BLOB);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_fr_array(&column_interpolation_poly, FIELD_ELEMENTS_PER_CELL);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_fr_array(&aggregated_interpolation_poly, FIELD_ELEMENTS_PER_CELL);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_bool_array(&is_cell_used, FIELD_ELEMENTS_PER_EXT_BLOB);
+    if (ret != C_KZG_OK) goto out;
+
+    // Initialize aggregated columns and usage flags
+    for (size_t i = 0; i < CELLS_PER_EXT_BLOB; i++) {
+        for (size_t j = 0; j < FIELD_ELEMENTS_PER_CELL; j++) {
+            size_t index = i * FIELD_ELEMENTS_PER_CELL + j;
+            aggregated_column_cells[index] = FR_ZERO;
+            is_cell_used[index] = false;
+        }
+    }
+
+    // Aggregate the cells by scaling and summing
+    for (size_t i = 0; i < num_cells; i++) {
+        for (size_t j = 0; j < FIELD_ELEMENTS_PER_CELL; j++) {
+            fr_t cell_fr, scaled_fr;
+            size_t offset = j * BYTES_PER_FIELD_ELEMENT;
+            ret = bytes_to_bls_field(&cell_fr, (const Bytes32 *)&cells[i].bytes[offset]);
+            if (ret != C_KZG_OK) goto out;
+            blst_fr_mul(&scaled_fr, &cell_fr, &r_powers[i]);
+            size_t index = cell_indices[i] * FIELD_ELEMENTS_PER_CELL + j;
+            blst_fr_add(
+                &aggregated_column_cells[index], &aggregated_column_cells[index], &scaled_fr
+            );
+
+            // Mark the cell as being used
+            is_cell_used[index] = true;
+        }
+    }
+
+    // Initialize aggregated interpolation polynomial
+    for (size_t i = 0; i < FIELD_ELEMENTS_PER_CELL; i++) {
+        aggregated_interpolation_poly[i] = FR_ZERO;
+    }
+
+    // Sum the interpolation polynomials
+    for (size_t i = 0; i < CELLS_PER_EXT_BLOB; i++) {
+        size_t index = i * FIELD_ELEMENTS_PER_CELL;
+
+        if (!is_cell_used[index]) continue;
+
+        ret = bit_reversal_permutation(
+            &aggregated_column_cells[index], sizeof(fr_t), FIELD_ELEMENTS_PER_CELL
+        );
+        if (ret != C_KZG_OK) goto out;
+
+        ret = fr_ifft(
+            column_interpolation_poly, &aggregated_column_cells[index], FIELD_ELEMENTS_PER_CELL, s
+        );
+        if (ret != C_KZG_OK) goto out;
+
+        uint64_t pos = reverse_bits_limited(CELLS_PER_EXT_BLOB, i);
+        fr_t inv_coset_factor;
+        blst_fr_eucl_inverse(&inv_coset_factor, &s->roots_of_unity[pos]);
+        shift_poly(column_interpolation_poly, FIELD_ELEMENTS_PER_CELL, &inv_coset_factor);
+
+        for (size_t k = 0; k < FIELD_ELEMENTS_PER_CELL; k++) {
+            blst_fr_add(
+                &aggregated_interpolation_poly[k],
+                &aggregated_interpolation_poly[k],
+                &column_interpolation_poly[k]
+            );
+        }
+    }
```

Comment:
You deleted a bunch of old comments. Can we add those back?

Reply:
Note, they don't have to be the same comments, just anything to help the reader.